src/elements/fields/FileUploadField.jsx

```diff
@@ -1,35 +1,218 @@
-import React from 'react';
+import React, { useRef } from 'react';
+import { useFileData, useThumbnailData } from '../../utils/image';
+import { justRemove } from '../../utils/array';
 
-import ReactForm from 'react-bootstrap/Form';
+import { FiX } from 'react-icons/fi';
+import { IconContext } from 'react-icons';
+import { Image } from 'react-bootstrap';
+import { FileUploadIcon } from '../components/icons';
 
-function FileUploadField({
+function RichFileUploadField({
   element,
-  fieldLabel,
   applyStyles,
   required = false,
   editable = false,
-  onChange = () => {},
-  onClick = () => {},
+  onChange: customOnChange = () => {},
+  onClick: customOnClick = () => {},
+  initialFiles = [],
   elementProps = {},
   children
 }) {
   const servar = element.servar;
+  const showLabel = servar.name !== '';
+  const isMultiple = servar.metadata.multiple;
+  const fileInput = useRef();
+
+  const [rawFiles, setRawFiles] = useFileData(initialFiles);
+  const thumbnailData = useThumbnailData(rawFiles);
+  // verify that thumbnail data still has length when there's no thumbnail for a text file
+  const allowMoreFiles = isMultiple || thumbnailData.length === 0;
+  const fileExists = thumbnailData.length > 0 || rawFiles.length > 0;
```

Comment:
why check both thumbnailData and rawFiles vs just rawFiles?

Reply:
the conditional used both in the previous multi upload field. I'll double check to see if I can identify a case where that's needed, otherwise theoretically it can be just rawFiles